src/celengine/completion.h

```diff
@@ -0,0 +1,36 @@
+// completion.h
+//
+// Copyright (C) 2024-present, the Celestia Development Team
+//
+// This program is free software; you can redistribute it and/or
+// modify it under the terms of the GNU General Public License
+// as published by the Free Software Foundation; either version 2
+// of the License, or (at your option) any later version.
+
+#pragma once
+
+#include <functional>
+#include <string>
+#include <celengine/selection.h>
+
+namespace celestia::engine
+{
+
+struct Completion
+{
+    Completion(const std::string& name, const Selection& selection, std::function<Selection()> resolver = nullptr) :
+        name(name),
+        selection(selection),
+        resolver(resolver)
+    {}
+
+    const std::string& getName() const;
+    const Selection& getSelection();
+
+private:
+    std::string name;
+    Selection selection;
```

Comment:
Possibly could be `std::variant<Selection, std::function<Selection()>>`

Reply:
switched to use std::variant instead and use it in constructor as well. I originally wanted to avoid `getSelection` calling resolver repeatedly so storing them separately. but it seems in our code it will only be called once.